sklearn/feature_selection/tests/test_mutual_info.py

```diff
@@ -207,3 +207,47 @@ def test_mutual_info_options(global_dtype):
         assert_allclose(mi_5, mi_6)
 
         assert not np.allclose(mi_1, mi_3)
+
+
+def test_mutual_information_symmetry_classif_regression(global_random_seed):
+    """Check that `mutual_info_classif` and `mutual_info_regression` are
+    symmetric by switching the target `y` as `feature` in `X` and vice
+    versa.
+
+    Non-regression test for:
+    https://github.com/scikit-learn/scikit-learn/issues/23720
+    """
+    rng = np.random.RandomState(global_random_seed)
+    n = 100
+    d, c = rng.randint(10, size=n), rng.normal(0, 1, size=n)
+
+    mi_classif = mutual_info_classif(
+        c[:, None], d, discrete_features=[False], random_state=123
```

Comment:
Nit:

```suggestion
        c[:, None], d, discrete_features=[False], random_state=rng
```

Reply:
rng would not work since we need it to be in the same state between the 2 calls. We can pass the global_random_seed instead